modular_zubbers/code/game/objects/items/fuel_flamer.dm

```diff
@@ -0,0 +1,125 @@
+
+/obj/item/gun/fuel_thrower
+	name = "fuel flamethrower"
+	icon_state = "flamethrowerbase"
+	inhand_icon_state = "flamethrower_0"
+	resistance_flags = FIRE_PROOF
+	icon = 'icons/obj/weapons/flamethrower.dmi'
+	lefthand_file = 'icons/mob/inhands/weapons/flamethrower_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/flamethrower_righthand.dmi'
+	dry_fire_sound = null
+	fire_sound = null
+	suppressed_sound = null
+	force = 3
+	throwforce = 10
+	throw_speed = 1
+	throw_range = 3
+	light_on = FALSE
+	light_system = OVERLAY_LIGHT
+	light_color = LIGHT_COLOR_FLARE
+	light_range = 2
+	light_power = 2
+	custom_materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT)
+	var/lit = FALSE
+	var/max_fuel = 100
+	var/fuel_consumption = 5
+	var/datum/reagent/fuel_type = /datum/reagent/fuel
+
+/obj/item/gun/fuel_thrower/Initialize(mapload)
+	. = ..()
+	desc = "A flamethrower that uses [lowertext(initial(fuel_type.name))] as ammunition."
+	chambered = new /obj/item/ammo_casing/flamer(src)
+	create_reagents(max_fuel)
+	reagents.add_reagent(fuel_type, max_fuel)
+	update_appearance(ALL)
+
+/obj/item/gun/fuel_thrower/update_icon_state()
+	inhand_icon_state = "flamethrower_[lit]"
```

Comment:
This adds 1 or 0 to the icon_state are you sure that#s what you want? i. e. flamethrower_0 and flamethrower_1

Reply:
this is using flamethrower sprites right now, which have those icon_states

Reply:
Cursed as heck, also sloppy to use the same inhands, but you get a pass for event code that's admin-spawn-only anyways.

Reply:
so i shouldn't use the same inhands for the same icon_state? and besides i update it to completely new sprites